aspnet/mvc/views/forms/sample/final/Controllers/HomeController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using FormsTagHelper.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH
+{
+    public class HomeController : Controller
+    {
+        public IActionResult Index()
+        {
+            var model = new CountryViewModel();
+            model.Country = "CA";
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Index(CountryViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                string message = model.Country +  " selected";
+                return RedirectToAction("IndexSuccess", new {id=message});
+            }
+
+            // If we got this far, something failed, redisplay form.
+            return View(model);
+        }
+
+        public IActionResult IndexMultiSelect()
+        {
+            var model = new CountryViewModelIEnumerable();
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult IndexMultiSelect(CountryViewModelIEnumerable model)
+        {
+            if (ModelState.IsValid)
+            {
+                string strCountriesSelected="";
+                foreach (string s in model.CountryCodes)
+                {
+                    strCountriesSelected = strCountriesSelected + " " + s;
+                }
+                return RedirectToAction("IndexSuccess", new { id = strCountriesSelected });
+            }
+
+            // If we got this far, something failed, redisplay form.
+            return View(model);
+        }
+
+        public IActionResult IndexGroup()
+        {
+            var model = new CountryViewModelGroup();
+           // model.Country = "FR";
```

Comment:
This and similar commented-out code has funky indentation. Why is the commented-out code in the sample at all?